src/devices/machine/qvt102.cpp

```diff
@@ -0,0 +1,692 @@
+	if (machine().phase() != machine_phase::INIT)
+	{
+		if (BIT(m_jumper->read(), 3))
+		{
+			fprintf(stderr, "######## rs232_write_dtr %02x\n", state);
+			m_rs232_conn_dtr_handler(state);
+		}
+	}
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_dcd_w)
+{
+	if (machine().phase() != machine_phase::INIT)
+		if (BIT(m_jumper->read(), 1))
+			m_acia->write_dcd(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_dsr_w)
+{
+	if (machine().phase() != machine_phase::INIT)
+		if (BIT(m_jumper->read(), 0))
+			m_acia->write_dcd(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_ri_w)
+{
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_cts_w)
+{
+	m_acia->write_cts(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_rxd_w)
+{
+	fprintf(stderr, "qvt102_device::rs232_rxd_w %02x\n", state);
+
+	m_acia->write_rxd(state);
+
+	// aux enable
+	if (BIT(m_latch, 0) == 0)
+		m_aux->write_txd(state);
+}
+
+void qvt102_device::device_resolve_objects()
+{
+	m_rs232_conn_dtr_handler.resolve_safe();
+	m_rs232_conn_rts_handler.resolve_safe();
+	m_rs232_conn_txd_handler.resolve_safe();
+}
+
+void qvt102_device::device_start()
+{
+	// register for save states
+	save_item(NAME(m_latch));
+	save_item(NAME(m_kbd_data));
+	save_item(NAME(m_kbd_bus));
+	save_item(NAME(m_kbd_p1));
+	save_item(NAME(m_kbd_p2));
+}
+
+void qvt102_device::device_reset()
+{
+	m_latch = 0;
+	m_kbd_data = 0;
+	m_kbd_bus = m_kbd_p1 = m_kbd_p2 = 0xff;
+}
+
+
+//**************************************************************************
+//  ROM DEFINITIONS
+//**************************************************************************
+
+ROM_START( qvt102 )
+	ROM_REGION(0x2000, "maincpu", 0)
+	ROM_LOAD("t205m.u8", 0x0000, 0x2000, CRC(59cc04f6) SHA1(ee2e3a3ea7b57a231483fcc74266f0f3f51204af))
+
+	ROM_REGION(0x1000, "chargen", 0)
+	ROM_LOAD("c3205m.u32", 0x0000, 0x1000, CRC(f6d86e87) SHA1(c0885e4a35095a730d760bf91a1cf4e8edd6a2bb))
+
+	ROM_REGION(0x400, "kbdmcu", 0)
+	ROM_LOAD("k301.u302",  0x000, 0x400, CRC(67564b20) SHA1(5897ff920f8fae4aa498d3a4dfd45b58183c041d))
+ROM_END
+
+ROM_START( qvt102a )
+	ROM_REGION(0x2000, "maincpu", 0)
+	ROM_LOAD("t205p.u8", 0x0000, 0x2000, CRC(2e375abc) SHA1(12ad1e49c5773c36c3a8d65845c9a50f9dec141f))
+
+	ROM_REGION(0x1000, "chargen", 0)
+	ROM_LOAD("c3205m.u32", 0x0000, 0x1000, CRC(f6d86e87) SHA1(c0885e4a35095a730d760bf91a1cf4e8edd6a2bb))
+
+	ROM_REGION(0x400, "kbdmcu", 0)
+	ROM_LOAD("k301.u302", 0x000, 0x400, CRC(67564b20) SHA1(5897ff920f8fae4aa498d3a4dfd45b58183c041d))
+ROM_END
+
+//**************************************************************************
+//  MACHINE DEFINTIONS
+//**************************************************************************
+
+void qvt102_device::device_add_mconfig(machine_config &config)
+{
+	M6800(config, m_maincpu, 16.6698_MHz_XTAL / 18);
+	m_maincpu->set_addrmap(AS_PROGRAM, &qvt102_device::mem_map);
+
+	// needs a tight sync with the keyboard cpu
+	config.m_perfect_cpu_quantum = subtag("maincpu");
```

Comment:
This should only ever be done in top-level machine configurations.  If it's allowed to be done from something that can be instantiated in a slot device, it could cause drivers to break in weird ways.  You'll have to find some other way to deal with synchronisation issues.